src/main/java/language/LanguageCommand.java

```diff
@@ -0,0 +1,383 @@
+
+                return new EmbedBuilder()
+                        .setTitle("Shortest Path to " + targetLanguageOpt.get())
+                        .setAuthor(user)
+                        .setDescription("The shortest path to " + targetLanguageOpt.get().getName() +
+                                " with this party is:\n" + joinPartyLinguisticsPaths(foundPaths)
+                        )
+                        .addField("Real Time Translation (Wits) / Grasp Idioms (Vision)", getGroupDifficultyString(foundPaths, 1))
+                        .addField("Translate Behind Conversation (Reason)", getGroupDifficultyString(foundPaths, 0))
+                        .setColor(RandomColor.getRandomColor());
+            }
+        }
+        else {
+            return new EmbedBuilder()
+                    .setTitle("Failed to Calculate Difficulty")
+                    .setDescription("No Target Language found for: " + option.getOptionStringValueByName("target").orElse("NONE"));
+        }
+    }
+
+    private String joinPartyLinguisticsPaths(Map<String, Try<List<Language>>> partyLanguageMap) {
+        return partyLanguageMap.entrySet().stream()
+                .map(stringListEntry -> String.format("%s: %s",
+                        stringListEntry.getKey(),
+                        stringListEntry.getValue()
+                                .fold(
+                                        Throwable::getMessage,
+                                        languages -> languages.stream()
+                                                .map(Language::getName)
+                                                .collect(Collectors.joining(" → "))
+                                )
+                ))
+                .collect(Collectors.joining("\n"));
+    }
+
+    private String getGroupDifficultyString(Map<String, Try<List<Language>>> partyLanguageMap, int modifier) {
+        return partyLanguageMap.entrySet().stream()
+                .map(stringListEntry -> String.format(
+                        "- %s: %s",
+                        stringListEntry.getKey(),
+                        stringListEntry.getValue()
+                                .fold(
+                                        Throwable::getMessage,
+                                        languages -> Tier.getByIndex(languages.size() + modifier)
+                                )
+                ))
+                .collect(Collectors.joining("\n"));
+    }
+
+    public static void setup(Velen velen, LanguageLogic languageLogic) {
+        VelenCommand.ofSlash(
+                "language",
+                "Query, Create, Connect, Remove Facets Languages",
+                velen,
+                new LanguageCommand(languageLogic),
+                new SlashCommandOptionBuilder()
+                        .setType(SlashCommandOptionType.SUB_COMMAND)
+                        .setName("add")
+                        .setDescription("Add a New Language")
```

Comment:
Can you add a period to the end of subcommands? The naming conventions for slash commands has periods at the end of commands, and no period at the end of parameters.

Reply:
Okay